src/mjolnir/ingest_transit.cc

```diff
@@ -389,33 +390,50 @@ bool write_stop_pair(Transit& tile,
   const auto& tile_tripId = feed_trip.id;
   const std::string currFeedPath = feed_trip.feed;
 
-  auto currTrip = feed.get_trip(tile_tripId);
+  const auto& currTrip = feed.get_trip(tile_tripId);
+  const auto& currShape = feed.get_shape(currTrip->shape_id);
+  const auto& trip_calendar = feed.get_calendar(currTrip->service_id);
```

Comment:
I moved these out of the stop_pair loop as they don't need to be computed every time